lib/awful/widget/layoutbox.lua

```diff
@@ -67,15 +112,49 @@ function layoutbox.new(args)
 
     screen = get_screen(screen or 1)
 
+    local uf = args.update_function or common.list_update
+    local w = base.make_widget_from_value(args.layout or wlayout.fixed.horizontal)
+
+    local data = setmetatable({}, { __mode = 'k' })
+
+    local queued_update = {}
+
+    function w._do_layoutbox_update_now()
+        if screen.valid then
+            update(w, screen, args.buttons, data, uf, args)
+        end
+        queued_update[screen] = false
+    end
+
+    function w._do_layoutbox_update()
+        -- Add a delayed callback for the first update.
```

Comment:
what is the idea behind it?
as far as i understand it, in case of multiple updates, it will take the first of them, not the last one

Reply:
Still working with it. I've mainly been looking at `tasklist` and `taglist` as templates. My answer right now: :man_shrugging:

Reply:
> what is the idea behind it?

For stuff like the taglist, calling `awful.widget.common` "update" and "label" is very expansive. Usually, more than 1 tag is created in the same event loop iteration. Without batching those calls, it slows down startup a lot. With the new client layout `append` API, the layoutbox will have the same problem, so I am happy to see batching being integrated so early! Usually it was the result of my profiling AwesomeWM and trying to address some bottlenecks.

Reply:
@Elv13 and what about second part of my question?

Reply:
I have one more question : `queued_update` is a local table. It means each instance of layoutbox will have its own `queued_update`. `screen` is also local to the layoutbox instance.

So, why don't we use a basic boolean variable, instead?

OR: the declaration of `queued_update` should be moved out of the constructor. (And probably use something else than the screen to identify the layoutbox instance, IMO.)